provider/src/index.ts

```diff
@@ -36,56 +36,94 @@ export default {
           'access-control-allow-methods': 'POST',
           'access-control-allow-headers': 'content-type',
           'access-control-max-age': '1800',
-          'access-control-allow-origin': request.headers.get('Origin'),
+          'access-control-allow-origin': '*',
           vary: 'Origin',
           'access-control-allow-credentials': 'true',
         },
       })
     }
 
+    const url = new URL(request.url)
+    const { pathname } = url
+    const proxyURL = url.searchParams.get('url')
+
+    if (pathname === '/proxy' && proxyURL) {
+      const endpoint = new URL(proxyURL)
+      // Proxy the request
+      const response = await fetch(endpoint.toString(), {
+        method: 'GET',
+        body: request.body,
+        headers: new Headers({
+          Accept: '*/*',
+          Origin: 'https://rpc.unlock-protocol.com/',
+          'Content-type': request.headers.get('content-type')!,
+        }),
+      })
+
+      const text = await response.text()
```

Comment:
What if we fail if the response is not a parse-able JSON with a `data` object in it?
This way we can make sure that it does not do anything useful for people who are not using it "correctly" ?

Reply:
Done.